mountpoint-s3-fs/examples/mount_from_config.rs

```diff
@@ -167,7 +172,29 @@ impl ConfigOptions {
     }
 }
 
+/// Reads the config_version field from a JSON config file and validates it against CONFIG_VERSION
+fn validate_config_version<P: AsRef<Path>>(path: P) -> Result<()> {
+    #[derive(Deserialize)]
```

Comment:
Would using [`from_str`](https://docs.rs/serde_json/latest/serde_json/fn.from_str.html) multiple times avoid needing to re-open the file?